core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.spec.ts

```diff
@@ -0,0 +1,306 @@
+        'SUBMITTED_QUESTION_SUGGESTIONS'
+      ).then(successHandler, failureHandler);
+
+      const req = http.expectOne(url);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(testSuggestionsBackendObject);
+
+      flushMicrotasks();
+
+      expect(carbas.fetchSubmittedSuggestionsAsync)
+        .toHaveBeenCalledWith('skill', 'add_question');
+      expect(successHandler).toHaveBeenCalled();
+      expect(failureHandler).not.toHaveBeenCalled();
+    }));
+
+    it('should fetch submitted translation suggestions', fakeAsync(() => {
+      spyOn(carbas, 'fetchSubmittedSuggestionsAsync').and.callThrough();
+      const url = '/getsubmittedsuggestions/exploration/translate_content';
+
+      carbas.fetchSuggestionsAsync(
+        'SUBMITTED_TRANSLATION_SUGGESTIONS'
+      ).then(successHandler, failureHandler);
+
+      const req = http.expectOne(url);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(testSuggestionsBackendObject);
+
+      flushMicrotasks();
+
+      expect(carbas.fetchSubmittedSuggestionsAsync)
+        .toHaveBeenCalledWith('exploration', 'translate_content');
+      expect(successHandler).toHaveBeenCalled();
+      expect(failureHandler).not.toHaveBeenCalled();
+    }));
+
+    it('should fetch reviewable question suggestions', fakeAsync(() => {
+      spyOn(carbas, 'fetchReviewableSuggestionsAsync').and.callThrough();
+      const url = '/getreviewablesuggestions/skill/add_question';
+
+      carbas.fetchSuggestionsAsync(
+        'REVIEWABLE_QUESTION_SUGGESTIONS'
+      ).then(successHandler, failureHandler);
+
+      const req = http.expectOne(url);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(testSuggestionsBackendObject);
+
+      flushMicrotasks();
+
+      expect(carbas.fetchReviewableSuggestionsAsync)
+        .toHaveBeenCalledWith('skill', 'add_question');
+      expect(successHandler).toHaveBeenCalled();
+      expect(failureHandler).not.toHaveBeenCalled();
+    }));
+
+    it('should fetch reviewable translation suggestions', fakeAsync(() => {
+      spyOn(carbas, 'fetchReviewableSuggestionsAsync').and.callThrough();
+      const url = '/getreviewablesuggestions/exploration/translate_content';
+
+      carbas.fetchSuggestionsAsync(
+        'REVIEWABLE_TRANSLATION_SUGGESTIONS'
+      ).then(successHandler, failureHandler);
+
+      const req = http.expectOne(url);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(testSuggestionsBackendObject);
+
+      flushMicrotasks();
+
+      expect(carbas.fetchReviewableSuggestionsAsync)
+        .toHaveBeenCalledWith('exploration', 'translate_content');
+      expect(successHandler).toHaveBeenCalled();
+      expect(failureHandler).not.toHaveBeenCalled();
+    }));
+  });
+
+  it('should correctly resolve suggestion to exploration', fakeAsync(() => {
+    const successHandler = jasmine.createSpy('success');
+    const failureHandler = jasmine.createSpy('failure');
+
```

Comment:
Can remove this line to group the test "setup" together. Here and below.

Reply:
done